clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedWithSSLTest.java

```diff
@@ -0,0 +1,141 @@
+
+@SpringBootTest(
+    classes = {JsonMapperConfiguration.class, ZeebeClientConfigurationImpl.class},
+    properties = {
+      "camunda.client.mode=self-managed",
+      "camunda.client.auth.client-id=my-client-id",
+      "camunda.client.auth.client-secret=my-client-secret",
+      "camunda.client.auth.keystore-password=password",
+      "camunda.client.auth.keystore-key-password=password",
+      "camunda.client.auth.truststore-password=password",
+    })
+@EnableConfigurationProperties({
+  ZeebeClientConfigurationProperties.class,
+  CamundaClientProperties.class
+})
+public class CredentialsProviderSelfManagedWithSSLTest {
+
+  private static final String VALID_TRUSTSTORE_PATH =
+      CredentialsProviderSelfManagedWithSSLTest.class
+          .getClassLoader()
+          .getResource("idp-ssl/truststore.jks")
+          .getPath();
+
+  private static final String VALID_IDENTITY_PATH =
+      CredentialsProviderSelfManagedWithSSLTest.class
+          .getClassLoader()
+          .getResource("idp-ssl/identity.p12")
+          .getPath();
+  private static final String VALID_CLIENT_PATH =
+      CredentialsProviderSelfManagedWithSSLTest.class
+          .getClassLoader()
+          .getResource("idp-ssl/localhost.p12")
+          .getPath();
+
+  @RegisterExtension
+  static WireMockExtension wm =
+      WireMockExtension.newInstance()
+          .options(
+              new WireMockConfiguration()
+                  .keystorePath(VALID_IDENTITY_PATH)
+                  .keystorePassword("password")
+                  .trustStorePath(VALID_TRUSTSTORE_PATH)
+                  .trustStorePassword("password")
+                  .dynamicHttpsPort())
+          .build();
+
+  private static final String ACCESS_TOKEN =
+      JWT.create().withExpiresAt(Instant.now().plusSeconds(300)).sign(Algorithm.none());
+
+  @MockBean ZeebeClientExecutorService zeebeClientExecutorService;
+  @Autowired ZeebeClientConfigurationImpl configuration;
+
+  @DynamicPropertySource
+  static void registerPgProperties(final DynamicPropertyRegistry registry) {
+    final String issuer = "https://localhost:" + wm.getHttpsPort() + "/auth-server";
+    registry.add("camunda.client.auth.issuer", () -> issuer);
+    registry.add("camunda.client.auth.keystore-path", () -> VALID_CLIENT_PATH);
+    registry.add("camunda.client.auth.truststore-path", () -> VALID_TRUSTSTORE_PATH);
+  }
+
+  @BeforeEach
+  void setUp() {
+    // Clean up credentials cache to ensure every test gets fresh token
+    Paths.get(System.getProperty("user.home"), ".camunda", "credentials")
+        .toAbsolutePath()
+        .toFile()
+        .delete();
+  }
+
+  @Test
+  void shouldBeSelfManaged() {
+    final CredentialsProvider credentialsProvider = configuration.getCredentialsProvider();
+    assertThat(credentialsProvider).isExactlyInstanceOf(OAuthCredentialsProvider.class);
+  }
+
+  @Test
+  void shouldHaveZeebeAuth() throws IOException {
```

Comment:
Hi @ana-vinogradova-camunda,

I am adding a new test case for identity provider only SSL support. I see a regular test class has the [same test disabled](https://github.com/camunda/camunda/blob/966f5c1b97a839da57c87757fb435cdd03be1004/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedTest.java#L94). I am wondering, why that test is disables and should I do the same? Should I even include it? Thank you!

Reply:
Ah, I see. Never observed it failing locally for both instances. I'll take a look if I can help with that